DigitalLearningSolutions.Web/ViewModels/LearningMenu/PostLearningContentViewModel.cs

```diff
@@ -0,0 +1,75 @@
+{
+    using System.Collections.Generic;
+    using DigitalLearningSolutions.Data.Models.PostLearningAssessment;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.Extensions.Configuration;
+
+    public class PostLearningContentViewModel
+    {
+        public int CustomisationId { get; }
+        public int SectionId { get; }
+        public string CourseTitle { get; }
+        public string SectionName { get; }
+        public string ContentSource { get; }
+        public List<int> Tutorials { get; }
+        private int CentreId { get; }
+        private int CandidateId { get; }
+        private int ProgressId { get; }
+
+        public PostLearningContentViewModel(
+            IConfiguration config,
+            PostLearningContent postLearningContent,
+            int customisationId,
+            int centreId,
+            int sectionId,
+            int progressId,
+            int candidateId
+        )
+        {
+            CustomisationId = customisationId;
+            CentreId = centreId;
+            SectionId = sectionId;
+            SectionName = postLearningContent.SectionName;
+            CandidateId = candidateId;
+            ProgressId = progressId;
+            CourseTitle = postLearningContent.CourseTitle;
+            Tutorials = postLearningContent.Tutorials;
+
+            ContentSource = ContentViewerHelper.IsScormPath(postLearningContent.PostLearningAssessmentPath)
+                ? GetScormSource(config, postLearningContent)
+                : GetHtmlSource(config, postLearningContent);
+        }
+
+        private string GetHtmlSource(
+            IConfiguration config,
+            PostLearningContent postLearningContent)
+        {
+            return $"{postLearningContent.PostLearningAssessmentPath}" +
+                   $"?CentreID={CentreId}" +
+                   $"&CustomisationID={CustomisationId}" +
+                   $"&CandidateID={CandidateId}" +
+                   $"&SectionID={SectionId}" +
+                   $"&Version={postLearningContent.Version}" +
+                   $"&ProgressID={ProgressId}" +
+                   "&type=pl" +
+                   $"&TrackURL={config.GetTrackingUrl()}" +
+                   $"&objlist=[{string.Join(",", Tutorials)}]" +
+                   $"&plathresh={postLearningContent.PassThreshold}";
+        }
+
+        private string GetScormSource(
+            IConfiguration config,
+            PostLearningContent postLearningContent)
+        {
+            return $"{config.GetScormPlayerUrl()}" +
+                   $"?CentreID={CentreId}" +
+                   $"&CustomisationID={CustomisationId}" +
+                   $"&CandidateID={CandidateId}" +
+                   $"&SectionID={SectionId}" +
+                   $"&Version={postLearningContent.Version}" +
+                   $"&tutpath={postLearningContent.PostLearningAssessmentPath}" +
+                   "&type=pl";
```

Comment:
Aren't these two very similar for diagnostic and post learning assessments? If so, worth extracting so the code can be shared? 🙂

Reply:
If we extract it there would still be a really big constructor in each class. If that's still an improvement then I'm happy to do so

Reply:
That's a good point. It would mainly be to ensure the two are consistent and the code to construct the url isn't duplicated. It's only two statements (one for scorm, one for non-scorm), but each is pretty complicated. I don't feel very strongly about it though, happy for you to keep it here if you prefer 🙂